devito/petsc/solver_parameters.py

```diff
@@ -0,0 +1,59 @@
+from petsctools import flatten_parameters
+import itertools
+
+"""
+Parameter descriptions:
```

Comment:
Can you attach this to 

linear_solver_parameters

so we can generate API doc for it